litt/littdb_config.go

```diff
@@ -0,0 +1,196 @@
+package litt
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/rand"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/litt/disktable/keymap"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/docker/go-units"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// Config is configuration for a litt.DB.
+type Config struct {
+	// The context for the database. If nil, context.Background() is used.
+	CTX context.Context
+
+	// The paths where the database will store its files. If the path does not exist, it will be created.
+	// If more than one path is provided, then the database will do its best to spread out the data across
+	// the paths. If the database is restarted, it will attempt to load data from all paths. Note: the number
+	// of paths should not exceed the sharding factor, or else data may not be split across all paths.
+	Paths []string
+
+	// The logger for the database. If nil, a logger is built using the LoggerConfig.
+	Logger logging.Logger
+
+	// The logger configuration for the database. Ignored if Logger is not nil.
+	LoggerConfig *common.LoggerConfig
+
+	// The type of the keymap. Choices are keymap.MemKeymapType and keymap.LevelDBKeymapType.
+	// Default is keymap.LevelDBKeymapType.
+	KeymapType keymap.KeymapType
+
+	// The default TTL for newly created tables (either ones with data on disk or new tables).
+	// The default is 0 (no TTL). TTL can be set individually on each table by calling Table.SetTTL().
+	TTL time.Duration
+
+	// The size of the control channel for the segment manager. The default is 64.
+	ControlChannelSize int
+
+	// The target size for segments. The default is math.MaxUint32.
+	TargetSegmentFileSize uint32
+
+	// The maximum number of keys in a segment. The default is 50,000. For workloads with moderately large values
+	// (i.e. in the kb+ range), this threshold is unlikely to be relevant. For workloads with very small values,
+	// this constant prevents a segment from accumulating too many keys. A segment with too many keys may have
+	// undesirable properties such as a very large key file and very slow garbage collection (since no kv-pair in
+	// a segment can be deleted until the entire segment is deleted).
+	MaxSegmentKeyCount uint64
+
+	// The desired maximum size for a key file. The default is 2 MB. When a key file exceeds this size, the segment
+	// will close the current segment and begin writing to a new one. For workloads with moderately large values,
+	// this threshold is unlikely to be relevant. For workloads with very small values, this constant prevents a key
+	// file from growing too large. A key file with too many keys may have undesirable properties such as very slow
+	// garbage collection (since no kv-pair in a segment can be deleted until the entire segment is deleted).
+	TargetSegmentKeyFileSize uint64
```

Comment:
Is this redundant with MaxSegmentKeyCount? Both are handling having many small values

Reply:
Similar, but not exactly the same. I allow there to be a limit placed on both the number of keys per segment, as well as the data size of the keys in a segment. Limiting both is important, since allowing too many keys or too much key data to accumulate can result in bad performance or memory pressure. Which threshold triggers before the other is workload dependent. If values are much larger than keys, it's very likely that neither threshold will ever get triggered (the segment is closed when the values grow too large).